server/src/main/java/org/elasticsearch/cluster/metadata/Metadata.java

```diff
@@ -1380,7 +1380,7 @@ public Builder removeTemplate(String templateName) {
         }
 
         public Builder templates(ImmutableOpenMap<String, IndexTemplateMetadata> templates) {
-            this.templates.putAll(templates);
+            this.templates.putAll((Map<String, IndexTemplateMetadata>) templates);
```

Comment:
The need for an explicit cast here is a little unfortunate. Is it disruptive to rename the builder method, to say putAllFromMap or something?